python/engine/PythonEngine.cpp

```diff
@@ -0,0 +1,170 @@
+#include "PythonEngine.hpp"
+#include <utilities/core/ApplicationPathHelpers.hpp>
+#include <fmt/format.h>
+#include <stdexcept>
+#include <string>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
+
+#include <Python.h>
+#include <SWIGPythonRuntime.hxx>
+
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+
+
+namespace openstudio {
+
+static constexpr auto importScript = R"(
+import sys
+sys.path.insert(0, '{}')
+import {}
+)";
```

Comment:
This looks weird to me to combine this + exec. It probably works fine.

But I was thinking it would be using the C API

```
PyObject* sys = PyImport_ImportModule( "sys" );
PyObject* sys_path = PyObject_GetAttrString( sys, "path" );
PyObject* folder_path = PyUnicode_FromString( "<insert folder path>" );
PyList_Append( sys_path, folder_path );
```

Reply:
Yeah I do feel a little guilty writing this in python. Would you feel better if I moved it over to C? 

I actually think PyConfig https://docs.python.org/3/c-api/init_config.html might be the most formal way to do this, but I looked at it and it seemed more complicated than I wanted.